### PR TITLE
Do not infer record fields if the LRH type is already known

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/3574 and https://github.com/Microsoft/visualfsharp/issues/3576

### CODE REVIEW

**Comment by vasily-kirichenko** [comment]
wait

**Comment by vasily-kirichenko** [comment]
OK, it's really great :) (I tried to use 
```fsharp
match adhoctDotSearchAccessible with 
    // If the type is already known, we should not try to lookup a record field
    | Exception _ when isRecdTy ncenv.g typ -> 
```
but it broke everything.

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@vasily-kirichenko breaking everything is a small price to pay for It's great ....

**Comment by vasily-kirichenko** [comment]
The current version seems to work OK. 

I'm not sure if this is the right solution because Don suggested to use `isNominalTy typ || isRecordTy` https://github.com/Microsoft/visualfsharp/issues/3576#issuecomment-329462185

**Comment by dsyme** [comment]
@vasily-kirichenko As I explained in https://github.com/Microsoft/visualfsharp/issues/3576#issuecomment-329462185 "isRecordTy" is not enough.  The type can be a type variable too (e.g. ``isTyparTy``, or, ``not isNominalTy`` is enough)

**Comment by dsyme** [comment]
@vasily-kirichenko Yes, what you have now should be enough

**Comment by cartermp** [comment]
@dotnet-bot test Windows_NT Release_ci_part2 Build please

**Comment by vasily-kirichenko** [comment]
it's ready

File: src/fsharp/NameResolution.fs [hunk 1]
```diff
@@ -3123,30 +3123,31 @@ let private ResolveExprDotLongIdent (ncenv:NameResolver) m ad nenv typ lid findF
...
-                    // NOTE (instantiationGenerator cleanup): we need to freshen here because we don't know the type. 
-                    // But perhaps the caller should freshen?? 
-                    let item = FreshenRecdFieldRef ncenv m rfref
-                    OneSuccess (ResolutionInfo.Empty,item,rest)
-            | _ -> NoResultsOrUsefulErrors 
+        let dotFieldIdSearch =
+            // If the type is already known, we should not try to lookup a record field
+            if isAppTy ncenv.g typ then 
+                NoResultsOrUsefulErrors
+            else match lid with
```
Comment (dsyme) [inline src/fsharp/NameResolution.fs]: Minor nit: this is odd code formatting for the compiler. We would normally use

```
    if X then 
        Y
    else
        match Z with 
```
or
```
    if X then Y else
    match Z with 
```

**Comment by vasily-kirichenko** [comment]
Done.

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test this please

**Pull request merged**